lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -134,6 +134,28 @@ def create(path)
       @zk.create(path, ignore: :node_exists)
     end
 
+    # get with retry to reduce failure when network connectivity is flaky or ZK is overloaded
+    def get(path, opts = {})
+      retry_limit = opts[:retry_limit] || 3
+      opts.delete(:retry_limit)
+      retry_interval = opts[:retry_interval] || 5
```

Comment:
`5` sec might be so long. We could try `1` `2` `3` respectively during three tries.

Reply:
To find reasonable retry interval, we have to experiment out. There is some trade-off. If retry interval is too small, we might make things worse by adding more load when zookeeper is already overloaded. If retry interval is too big, it obviously delay the time to recover from intermittent network connectivity issue.

You are right, exponential back-off is probably a more elegant solution. Before throwing sophisticated solution here, I want to try out something simple first.